runtime/syntax/synload.vim

```diff
@@ -48,8 +48,10 @@ fun! s:SynSet()
     " load each in sequence.  Skip empty entries.
     for name in split(s, '\.')
       if !empty(name)
-        exe "runtime! syntax/" . name . ".vim syntax/" . name . "/*.vim"
-        exe "runtime! syntax/" . name . ".lua syntax/" . name . "/*.lua"
+        exe printf(
+          \ "runtime! syntax/%s.vim syntax/%s/*.vim  syntax/%s.lua syntax/%s/*.lua",
+          \ name, name, name, name
+          \ )
```

Comment:
Just to be clear: the ordering "Vim first, then Lua" is now fully taken care of by `runtime!`? Otherwise you need to interleave Vim and Lua files as in #24147 

(also check `ftplugin.vim` and `indent.vim`, cf. #20322)

Reply:
Oh, and do these changes also apply to `nvim__get_runtime`? Could try using this instead of `runtime!` here, then.

Reply:
Yes I believe so. Any pattern passed to `runtime` will source vim files first.

Reply:
...but making sure to respect directory ordering

Reply:
Not, this is now the same style that `ftplugin.vim` does not **currently**. Not sure if that was intentional:

```vim
      " When there is a dot it is used to separate filetype names.  Thus for
      " "aaa.bbb" load "aaa" and then "bbb".
      for name in split(s, '\.')
        " Load Lua ftplugins after Vim ftplugins _per directory_
        " TODO(clason): use nvim__get_runtime when supports globs and modeline
        exe printf('runtime! ftplugin/%s.vim ftplugin/%s.lua', name, name)
        exe printf('runtime! ftplugin/%s_*.vim ftplugin/%s_*.lua', name, name)
        exe printf('runtime! ftplugin/%s/*.vim ftplugin/%s/*.lua', name, name)
      endfor
```

Need to double-check how `:runtime` handles each argument; does each argument do a seperate scan(?).

Reply:
Ok, currently each argument to `:runtime!` performs a separate scan, so the change in this file shouldn't do anything.

```vim
:runtime! a
:runtime! b
```
should be the same as
```
runtime! a b
```

Reply:
> Note, this is now the same style that `ftplugin.vim` does **currently**. Not sure if that was intentional:
> 
> ```viml
>       " When there is a dot it is used to separate filetype names.  Thus for
>       " "aaa.bbb" load "aaa" and then "bbb".
>       for name in split(s, '\.')
>         " Load Lua ftplugins after Vim ftplugins _per directory_
>         " TODO(clason): use nvim__get_runtime when supports globs and modeline
>         exe printf('runtime! ftplugin/%s.vim ftplugin/%s.lua', name, name)
>         exe printf('runtime! ftplugin/%s_*.vim ftplugin/%s_*.lua', name, name)
>         exe printf('runtime! ftplugin/%s/*.vim ftplugin/%s/*.lua', name, name)
>       endfor
> ```
> 
> Need to double-check how `:runtime` handles each argument; does each argument do a seperate scan(?).

No, the current state of `ftplugin.vim` is **wrong** (I messed up); #24147 should be the right behavior (I hope; still waiting for confirmation).